src/colvarbias_restraint.h

```diff
@@ -280,6 +280,9 @@ class colvarbias_restraint_harmonic_walls
   /// \brief If both walls are defined, use this k for the upper
   cvm::real upper_wall_k;
 
+  /// \brief Whether add biases on the extended DOF
+  bool b_bias_on_extended_value;
```

Comment:
@fhh2626 Why not make it part of the `colvarbias` base class?

Reply:
I agree.

Reply:
This is not on the shared repository, so I probably can't push to it.

Reply:
Better yet, this should be a feature, not a boolean flag.  Please leave as is until merged.

Reply:
Sorry, I'm a little bit confused. Whenever a extended DOF is added into the system, it is always implied that the biases are added on the extended DOF. `colvarbias_restraint_harmonic_walls` is the an exception, isn't it? Then, what does it mean if `b_bias_on_extended_value` is part of `colvarbias`?

Reply:
What you are proposing is adding a keyword that allows the user to control this behavior in a way that is general to many classes: it makes sense for consistency to have it available everywhere. The default value for this new feature could be `true` for the for most biases, and `false` for `harmonicWalls` specifically.

If a new restraint in the future needs to behave the same way as `harmonicWalls`, this can be achieved by making the default value in the same class `false` as well (as opposed to duplicating the relevant code).

Having a `feature` instead of a simple `bool` flag also enables to do internal error checks at run time; but features are defined at the base-class level (and may or may not be used by the derivative classes).